prime-router/docs/features/0004-managing-translation-schemas.md

```diff
@@ -0,0 +1,199 @@
+# Managing Translation Schemas
+
+## Context
+
+Schemas are used at multiple points in the universal pipeline
+
+- transforming items sent by senders
+- transforming items right before they are delivered to a receiver
+
+and cover different kinds of transforms:
+
+- HL7 -> FHIR
+    - **This is handled by another [library](https://github.com/LinuxForHealth/hl7v2-fhir-converter) with limited configuration option of where to read the mapping files from**
+- FHIR -> FHIR
+- FHIR -> HL7
+
+These schemas are currently stored as files that are included in the deployed application and read from the file sytem
+when they are applied to an item.
+
+## Problem
+
+The overarching problem is that since the schemas are part of the deployed application they can only be updated at the frequency at which the application itself is deployed.  This is quite limiting and downstream impacts such as:
+
+- onboarding new senders/receivers can take a while as iterative changes to the schema have to tested out over several days or weeks
+- bugs discovered in the schema cannot be immediately addressed without a hotfix
+- the senders/receivers do not have any capability to self-serve changes to their own schemas
+
+## Storing and resolving schemas
+
+### DB
+
+This solution would involve a drastic re-write from the existing functionality as the current libraries are all driven by the file system and the solution would likely involve moving away from schemas that could be edited by hand.
```

Comment:
I don't think we need a re-write. We're planning to migrate the FHIR to HL7 library to its own repo and allow it to be used externally. I think we should keep the functionality to read from YAML files and not make it dependent on configuring a database if people plan to use this library.